e2e/ruler_test.go

```diff
@@ -254,6 +254,7 @@ alerting:
     alertmanagers:
         - follow_redirects: true
           enable_http2: true
+          http_headers: null
```

Comment:
I wished we could switch to https://github.com/GoogleCloudPlatform/prometheus-engine/issues/1629 otherwise we have this small rolling restart risk (operator upgrades, but ruler/collector not yet). 

Since you want to put it in 0.15 looks like it won't happen and we have to live with that risk.